llama-index-integrations/embeddings/llama-index-embeddings-huggingface/llama_index/embeddings/huggingface/base.py

```diff
@@ -45,93 +42,70 @@ class HuggingFaceEmbedding(BaseEmbedding):
         description="Instruction to prepend to text."
     )
     cache_folder: Optional[str] = Field(
-        description="Cache folder for huggingface files."
+        description="Cache folder for Hugging Face files."
     )
 
     _model: Any = PrivateAttr()
-    _tokenizer: Any = PrivateAttr()
     _device: str = PrivateAttr()
 
     def __init__(
         self,
-        model_name: Optional[str] = None,
-        tokenizer_name: Optional[str] = None,
-        pooling: Union[str, Pooling] = "cls",
+        model_name: str = DEFAULT_HUGGINGFACE_EMBEDDING_MODEL,
+        tokenizer_name: Optional[str] = "deprecated",
+        pooling: str = "deprecated",
```

Comment:
@tomaarsen Hello, I am trying out `HuggingFaceEmbedding(model_name='jinaai/jina-embeddings-v2-base-en', pooling='mean', trust_remote_code=True)` and I got `deprecated` error. I wonder how can I do `mean pooling` now? Thanks.

Reply:
You can leave out the `pooling` parameter. Mean pooling is now used automatically based on this Sentence Transformers config file: https://huggingface.co/jinaai/jina-embeddings-v2-base-en/blob/main/1_Pooling/config.json

- Tom Aarsen

Reply:
@tomaarsen Thanks Tom for your time and help. That is very strange. I was evaluating different retrievers and got the surprisingly low result for jina embedding. That is why I was suspecting that mean pooling was not used, as it was supposed to similar performance according to the notebook from the blog '[Boosting RAG: Picking the Best Embedding & Reranker models](https://www.llamaindex.ai/blog/boosting-rag-picking-the-best-embedding-reranker-models-42d079022e83)'. Thanks.

<img width="802" alt="image" src="https://github.com/user-attachments/assets/80caee53-f560-4d6a-b0b4-60c8f213e61a">